app/scripts/comp/launcher-electron.js

```diff
@@ -0,0 +1,266 @@
+'use strict';
+
+const Backbone = require('backbone');
+const Locale = require('../util/locale');
+const Logger = require('../util/logger');
+
+const logger = new Logger('launcher');
+
+const Launcher = {
+    name: 'electron',
+    version: window.process.versions.electron,
+    autoTypeSupported: true,
+    req: window.require,
+    platform: function() {
+        return process.platform;
+    },
+    electron: function() {
+        return this.req('electron');
+    },
+    remoteApp: function() {
+        return this.electron().remote.app;
+    },
+    remReq: function(mod) {
+        return this.electron().remote.require(mod);
+    },
+    openLink: function(href) {
+        this.electron().shell.openExternal(href);
+    },
+    devTools: true,
+    openDevTools: function() {
+        this.electron().remote.getCurrentWindow().openDevTools();
+    },
+    getSaveFileName: function(defaultPath, cb) {
+        if (defaultPath) {
+            const homePath = this.remReq('electron').app.getPath('userDesktop');
+            defaultPath = this.req('path').join(homePath, defaultPath);
+        }
+        this.remReq('electron').dialog.showSaveDialog({
+            title: Locale.launcherSave,
+            defaultPath: defaultPath,
+            filters: [{ name: Locale.launcherFileFilter, extensions: ['kdbx'] }]
+        }, cb);
+    },
+    getUserDataPath: function(fileName) {
+        return this.req('path').join(this.remoteApp().getPath('userData'), fileName || '');
+    },
+    getTempPath: function(fileName) {
+        return this.req('path').join(this.remoteApp().getPath('temp'), fileName || '');
+    },
+    getDocumentsPath: function(fileName) {
+        return this.req('path').join(this.remoteApp().getPath('documents'), fileName || '');
+    },
+    getAppPath: function(fileName) {
+        return this.req('path').join(this.remoteApp().getAppPath(), fileName || '');
+    },
+    getWorkDirPath: function(fileName) {
+        return this.req('path').join(process.cwd(), fileName || '');
+    },
+    writeFile: function(path, data, callback, error) {
+        try {
+            this.req('fs').writeFileSync(path, new window.Buffer(data));
```

Comment:
To convert sync to async, it makes sense to use node.js async methods. Otherwise, we end up with calling sync code in fake-async way.
❗️ Changed to async methods.